src/components/HeaderNavMenu.jsx

```diff
@@ -21,6 +26,23 @@ function HeaderNavMenu() {
     About: '/about',
     Contact: '/contact',
   };
+
+  const [isSignedIn, setIsSignedIn] = useState(!!firebase.auth().currentUser);
+  const [loading, setLoading] = useState(true);
+
+  function updateUserStatus(user) {
+    setLoading(false);
+    setIsSignedIn(!!user);
+  }
+
+  useEffect(
+    () => firebase.auth().onAuthStateChanged((user) => updateUserStatus(user)),
+    []
+  );
+
```

Comment:
While you're moving this around, a better idea around this is to initialize `isSignedIn` to `undefined` by default like we do in [ProtectedRoute](https://github.com/beyondhb1079/s4us/blob/e85ef2def10365a1fcd91d7d20e0fc7b5b8779e6/src/components/ProtectedRoute.jsx) and set `loading` to true if `isSignedIn` is `undefined`:

```js
  const [isSignedIn, setIsSignedIn] = useState(!!firebase.auth().currentUser || undefined);
  const loading = isSignedIn == undefined;

  useEffect(
    () => firebase.auth().onAuthStateChanged((user) => setIsSignedIn(!!user)),
    []
  );
```

This makes sure we know when we're in an undefined state vs. the yes/no states afterwards.

Reply:
Yeah this make a whole lot of sense to do. The changes are made